File: ms-connect/src/main/java/com/h_salvacao/ms_connect/entity/Receita.java
package com.h_salvacao.ms_connect.entity;

import com.h_salvacao.ms_connect.util.Medicacao;
import com.h_salvacao.ms_connect.util.Queue;
import com.h_salvacao.ms_connect.util.QueueConverter;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "tb_receitas")
public class Receita {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String numToken;
    private Long fichaId;
    private String nomePaciente;
    @Convert(converter = QueueConverter.class)
    private Queue<Medicacao> medicacoes;



}

File: ms-token-app/src/app/entity/receita.ts
import { Token } from "./token";
import { Ficha } from "./ficha";
import { Medicacao } from "./medicacao";

export class Receita{
    id: number;
    numToken: string;
    fichaId: number;
    nomePaciente: String;
    medicacoes: Medicacao[];
}